features/ALS016_Alignment-horizontal-segment-geometric-continuity.feature

```diff
@@ -0,0 +1,29 @@
+@industry-practice
+@ALS
+@version1
+
+Feature: ALS016 - Alignment horizontal segment geometric continuity
+
+  The rule verifies that there is geometric continuity between segments in an IfcCompositeCurve.
+  The calculated end position and tangent vector of segment `n` is compared to the provided placement of segment `n + 1`.
+  A warning is emitted if the calculated difference is greater than the specified tolerance of the applicable
+  geometric context.
+
+Background:
+
+  Given A model with Schema "IFC4.3"
+  Given An IfcAlignment
+  Given Its attribute Representation
+  Given Its attribute Representations
+  Given RepresentationType = 'Curve2D'
+  Given All referenced instances
+  Given Its Entity Type is 'IfcCompositeCurve'
+  Given Its attribute Segments grouped pairwise
```

Comment:
```suggestion
  Given Its attribute Segments 
  Given The values grouped pairwise
```

I think I would have preferred this broken down into two statements, but we can revisit that when we have multiple grouping operations or grouping operations on in a different context instead of attribute retrieval.

The alternative is to supply a utility function used in the implementation of the then function to lookup the next sibling in execution context, but I think this approach in the given statements is very nice.

Reply:
I will go ahead and make the change now as I anticipate `pairwise` being re-used very soon for things like confirming same number of vertices on cross-sections being used for swept geometry.

Reply:
Cool. Thanks for taking the time. I just realized I forgot another point related to the complete lack of separation of concerns in the IFC schema.

What if we have an IfcAlignment that uses IfcCompositeCurveSegment as opposed to IfcCurveSegment (lord, do I hate this naming distinction). I.e the pre 4.3 composite curve with regular step segments. We at least need to make sure this is gracefully handled, preferably explicitly.

I think Alignment with pre-4.3  composite curve should be allowed as wel also allow polyline.

Maybe something we can do something with `Given its entity type is IfcCurveSegment`. But my early morning brain cannot predict how that would work with pairwise.

That would also mean `ALS011 - Alignment segment entity type consistency` is kind of a precursory check for this rule.

Thoughts?

Reply:
Agree.  I added `Its Entity Type is 'IfcCurveSegment'`